src/app/feature/campaign/campaign.service.ts

```diff
@@ -306,16 +318,34 @@ export class CampaignService {
   }
 
   private evaluateCampaignNotification(scheduleRow: FlowTypes.Campaign_Schedule) {
+    // apply default settings
+    scheduleRow.time = scheduleRow.time || NOTIFICATION_DEFAULTS.time;
     const { time, delay } = scheduleRow;
     const schedule: FlowTypes.Campaign_Schedule["schedule"] = scheduleRow.schedule || {};
-    // set a base date from today or schedule start (if provided)
-    let d = schedule.start_date ? new Date(schedule?.start_date) : new Date();
+
+    let d = new Date();
+    // Set notification start date (if in future, otherwise keep as today)
+    if (schedule.start_date && isFuture(new Date(schedule.start_date))) {
+      d = new Date(schedule.start_date);
+    }
+    // Set notification time, add 1 day if time already passed
+    d.setHours(Number(time.hour));
+    d.setMinutes(Number(time.minute));
+    if (isPast(d)) {
+      d = addDays(d, 1);
+    }
+    // Schedule notification day of week, add 1 week if time already passed
     if (schedule.day_of_week) {
       d = setISODay(d, schedule.day_of_week);
```

Comment:
@chrismclarke This is the one bit I don't know if I fully understand can this go backwards? If so do we need to check against the start date again? If not why are we checking against the past again? 
Worst case scenario is that we may have a rare bug where it sends a reminder before the start date, which is not a serious problem at the moment and would be easy to catch so I will merge anyway.

Reply:
I think everything should be forward moving. 

The start date is today (now), and we move it forward iff the specified start_date is in the future (we still want to schedule for the current day if we are in between start/end dates):
```
let d = new Date(); // (today)
if (schedule.start_date && isFuture(new Date(schedule.start_date))) {
      d = new Date(schedule.start_date); // (future start date)
    }
```
Then when it comes to setting time we check against the past because if it is currently 1pm and I'm scheduling for 7am, I could be scheduling for earlier in the same day (in which case we add a day to make it tomorrow at 7am).
```
    d.setHours(Number(time.hour));
    d.setMinutes(Number(time.minute));
    if (isPast(d)) {
      d = addDays(d, 1); // (set date one day later if the time scheduled is in past, i.e. earlier in same day)
    }
``` 
We do similarly for day_of_week, e.g. if it is currently Wednesday and we are setting for Monday it will default to Monday of current week (which is in the past), so we add on a week
```
if (schedule.end_date) {
      if (isAfter(d, endOfDay(new Date(schedule.end_date)))) {
        return null;
      }
    }
```
At least that's how things should be working, may be a case I missed somewhere...